### ISSUE TITLE
Front end crashes with an exception while loading a dill file produced from a compilation that has errors

### ISSUE THREAD

**Issue by a-siva** [dart-lang/sdk#32384]
The front end crashes with the following stack trace
```
compiler message: Unhandled exception:
package:flutter/widgets.dart: Internal problem: Unhandled Null in finalizeExports.
#0      internalProblem (package:front_end/src/fasta/problems.dart:31)
#1      unhandled (package:front_end/src/fasta/problems.dart:44)
#2      DillLibraryBuilder.finalizeExports (package:front_end/src/fasta/dill/dill_library_builder.dart:185)
#3      DillLoader.finalizeExports.<anonymous closure> (package:front_end/src/fasta/dill/dill_loader.dart:71)
#4      _HashVMBase&MapMixin&&_LinkedHashMapMixin.forEach (dart:collection-patch/dart:collection/compact_hash.dart:361)
#5      DillLoader.finalizeExports (package:front_end/src/fasta/dill/dill_loader.dart:69)
#6      DillTarget.buildOutlines (package:front_end/src/fasta/dill/dill_target.dart:54)
<asynchronous suspension>
#7      IncrementalCompiler.computeDelta.<anonymous closure> (package:front_end/src/fasta/incremental_compiler.dart:123)
<asynchronous suspension>
#8      CompilerContext.runInContext.<anonymous closure> (package:front_end/src/fasta/compiler_context.dart:103)
#9      _rootRun (dart:async/zone.dart:1126)
#10     _CustomZone.run (dart:async/zone.dart:1023)
```

To reproduce:

run the flutter gallery app in --preview-dart-2 mode
make some changes in a flutter packages file with a change that will trigger a compilation errors and hot reload the app. Now quit the app and then restart it again. The second time around when reading the dill file it produces the above stack trace.

**Comment by a-siva** [comment]
see https://github.com/flutter/flutter/issues/15033 for the corresponding flutter issue

**Comment by jensjoha** [comment]
A workaround is about to land.
I will dig into the underlying problem next week.

**Comment by a-siva** [comment]
This seems to happen every time the app is quit  (current flutter run session) and a new session is initiated using 'flutter run' on the app. Bumping priority to P0.

**Comment by jensjoha** [comment]
Should have been fixed (worked around) by 57a01968ed85eb25cf5fe02fe3b64f2776d8f085

**Comment by jensjoha** [comment]
The underling problem will be fixed by https://dart-review.googlesource.com/c/sdk/+/44901
